src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyTests.Fixer.cs

```diff
@@ -29,5 +30,186 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpDoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyFixer();
         }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstExtensionCallCSharp()
+        {
+            VerifyCSharpFix(@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list.First();
+        var f2 = GetList().First();
+        var f3 = matrix[0].First();
+        Console.WriteLine(list.First());
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list[0];
+        var f2 = GetList()[0];
+        var f3 = matrix[0][0];
+        Console.WriteLine(list[0]);
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", allowNewCompilerDiagnostics: true);
```

Comment:
What new diagnostic is generated? Can it be avoided?

Reply:
@mavasani `CS8019: Unnecessary using directive.` It's complaining about `System.Linq` since there's nothing else using it. I think it's pretty harmless.

Reply:
Can you add pragma disable for it in source instead of blanket disable all diagnostics in new code?

Reply:
Will do.